include/deal.II/numerics/data_postprocessor.h

```diff
@@ -591,8 +595,169 @@ class DataPostprocessorScalar : public DataPostprocessor<dim>
+ *                                   update_gradients)
+ *   {}
+ *
+ *   virtual
+ *   void
+ *   evaluate_scalar_field (const DataPostprocessorInputs::Scalar<dim> &input_data,
+ *                          std::vector<Vector<double> >               &computed_quantities) const
+ *   {
+ *     // ensure that there really are as many output slots
+ *     // as there are points at which DataOut provides the
+ *     // gradients:
+ *     AssertDimension (input_data.solution_gradients.size(),
+ *                      computed_quantities.size());
+ *
+ *     // then loop over all of these inputs:
+ *     for (unsigned int p=0; p<input_data.solution_gradients.size(); ++p)
+ *       {
+ *         // ensure that each output slot has exactly 'dim'
+ *         // components (as should be expected, given that we
+ *         // want to create vector-valued outputs), and copy the
+ *         // gradients of the solution at the evaluation points
+ *         // into the output slots:
+ *         AssertDimension (computed_quantities[p].size(), dim);
+ *         for (unsigned int d=0; d<dim; ++d)
+ *           computed_quantities[p][d]
+ *             = input_data.solution_gradients[p][d];
+ *       }
+ *   }
+ * };
+ * @endcode
+ * The only thing that is necessary is to add another output to the call
+ * of DataOut::add_vector() in the @p run() function of the @p Step6 class
+ * of that example program. The corresponding code snippet would then look
+ * like this (where we also use VTU as the file format to output the data):
+ * @code
+ *   GradientPostprocessor<dim> gradient_postprocessor;
+ *
+ *   DataOut<dim> data_out;
+ *   data_out.attach_dof_handler (dof_handler);
+ *   data_out.add_data_vector (solution, "solution");
+ *   data_out.add_data_vector (solution, gradient_postprocessor);
+ *   data_out.build_patches ();
+ *
+ *   std::ofstream output ("solution.vtu");
+ *   data_out.write_vtu (output);
+ * @endcode
+ *
+ * This leads to the following output for the solution and the
+ * gradients (you may want to compare with the solution shown in the results
+ * section of step-6; the current data is generated on a coarser mesh for
+ * simplicity):
+ *
+ * @image html data_postprocessor_vector_0.png
+ * @image html data_postprocessor_vector_1.png
+ *
+ * In the second image, the background color corresponds to the magnitude of the
+ * gradient vector and the vector glyphs to the gradient itself. It may be surprising
+ * at first to see that from each vertex, multiple vectors originate, going in
+ * different directions. But that is because the solution is only continuous:
+ * in general, the gradient is discontinuous across edges, and so the multiple
+ * vectors originating from each vertex simply represent the differing
+ * gradients of the solution at each adjacent cell.
+ *
+ * The output above -- namely, the gradient $\nabla u$ of the solution --
+ * corresponds to the "strain" in linear elasticity if the solution were scalar.
```

Comment:
I think that relating this to linear elasticity is more confusing than it is helpful. Can't we rather use a physical scalar field, e.g. temperature, that offers a much more intuitive understanding of the results and later discussion?